sys/net/skald/skald_eddystone.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_skald_eddystone
+ * @{
+ *
+ * @file
+ * @brief       Skald's Eddystone implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "assert.h"
+#include "net/skald/eddystone.h"
+
+#define PREAMBLE_LEN            (11U)
+#define PA_LEN                  (7U)
+#define PB_LEN                  (3U)
+
+#define URL_HDR_LEN             (6U)
+
+#define UID_LEN                 (23U)
+
+typedef struct __attribute__((packed)) {
+    uint8_t txadd[BLE_ADDR_LEN];
+    uint8_t pa[PA_LEN];
+    uint8_t service_data_len;
+    uint8_t pb[PB_LEN];
+    uint8_t type;
+} pre_t;
+
+typedef struct __attribute__((packed)) {
+    pre_t pre;
+    uint8_t tx_pwr;
+    uint8_t namespace[EDDYSTONE_NAMESPACE_LEN];
+    uint8_t instance[EDDYSTONE_INSTANCE_LEN];
+    uint8_t reserved[2];
+} eddy_uid_t;
+
+typedef struct __attribute__((packed)) {
+    pre_t pre;
+    uint8_t tx_pwr;
+    uint8_t scheme;
+    uint8_t url;
```

Comment:
The URL is only one byte long? oO

Reply:
Nope, but this field is only used for its address.

Reply:
Maybe use `url[]` instead then to make this clearer. (additional benefit, you don't need the address operator `&` to get the pointer ;-)).